scripts/potoken_provider/setup_pot_provider.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -e
+
+SCRIPTS_DIR="scripts/potoken_provider"
+TARGET_DIR="$SCRIPTS_DIR/bgutil-provider"
+GEN_SCRIPT="$TARGET_DIR/build/generate_once.js"
+
+# Ensure the server directory exists
+if [ ! -d "$TARGET_DIR" ]; then
+    echo "Error: PO Token provider server directory is missing! Please run scripts/update_pot_provider.sh first."
+    exit 1
+fi
+
+cd "$TARGET_DIR" || exit 1
+
+# Check if dependencies need installation
+if [ ! -d "node_modules" ]; then
```

Comment:
You should do a `yarn install` regardless of `node_module`'s existence. Sometimes Youtube will break things for us, and we need to update our main dependency (youtubei.js) to fix it. 

If no dependencies have changed, the `yarn install` is a no-op, so you won't be wasting any time on this regardless